Add packages that have always been MIT

ljharb opened the pull request:
All of these packages are MIT, but not SPDX-compliantly declared, until the version number listed.

- [x] Downloaded, extracted, and reviewed the contents of the package tarball on the npm public registry.
- [x] Checked `README`, `LICENSE`, `COPYING`, and header comments for conflicting or additional licenses.
- [x] Compared the license text found to standard language on <htts://spdx.org/licenses/>.
- [ ] Asked the package author about the applicable license.

^ seems not necessary, since they've always declared MIT, just not SPDX-compliantly.

I used semver notation here because it seems unnecessarily redundant to declare every explicit version, but if needed I can programmatically expand each of these with some creative `npm show` shell commands.

ljharb committed:
Add packages that have always been MIT

All of these packages are MIT, but not SPDX-compliantly declared, until the version number listed.

kemitchell commented:
@ljharb we need specific versions. The registry doesn't allow different tarballs for the same name@version, but it doesn't stop folks from going back and, say, doing patch releases on old major versions.

We need to review every build artifact to be sure of the license terms.

ljharb commented:
Updated.

kemitchell commented:
@ljharb Do you mind giving me just a capsule summary of how you went about auditing all these?

ljharb commented:
@kemitchell I looked at the git history for each of their package.jsons, and confirmed that at every version prior to the SPDX-compliant license string being added (enhanced-resolve is the only one that hadn't added it yet), the MIT license was being declared, and then used unpkg.com to explore a sampling of the build artifacts for them to confirm that they always matched the git history (for package.json, it's *exceedingly* rare that this will ever differ).

I haven't exhaustively checked each one of these on unpkg, only the earliest, latest, latest of each major, and first of each major, but i think if that's going to be a requirement, it'd be better to build a tool to do that for us rather than require exhaustive human research.

kemitchell commented:
@ljharb I'm all for building tools. We do need verification of each build artifact we code here. Otherwise it's not clear what folks are relying on.

ljharb commented:
Isn't the declaration of intent in the git repo sufficient? (I assume it's not been tested in court either way)

kemitchell commented:
@ljharb Git repos don't reliably match build artifacts in the registry. They _usually_ do. But there are discrepancies.

ljharb commented:
Right, but the issue with licensing isn't just about technical bit matching, isn't it? If I declare it's MIT in the repo but sneakily change the build artifact to be GPL, would any court consider that GPL?

kemitchell commented:
@ljharb Do you want that problem?

ljharb commented:
lol I want someone to have it, so precedent can be established and we don't have to waste our time unnecessarily :-) but alright. i'll see about verifying each build artifact using unpkg.

ljharb commented:
Here's the results for every version in index.csv after this PR lands:

kemitchell commented:
@ljharb Forgot to paste?

Back on tools: https://github.com/kemitchell/dotfiles-nodejs/blob/48c06aee3c2772f0f07395c22b2853d51b18d1d5/local/bin/audit-npm-package

kemitchell commented:
@ljharb drop yargs. I can ask Ben.

ljharb commented:
yargs is already fixed in newer versions; these are corrections for older ones.

I just submitted the comment too soon; it's coming, one sec.

ljharb commented:
Here's what I got from my script:
<details>

```sh
acorn-jsx@0.6.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://marijnhaverbeke.nl/acorn/LICENSE"   } ] 
acorn-jsx@1.0.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://raw.githubusercontent.com/RReverser/acorn-jsx/master/LICENSE"   } ] 
acorn-jsx@1.0.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://raw.githubusercontent.com/RReverser/acorn-jsx/master/LICENSE"   } ] 
acorn-jsx@1.0.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://raw.githubusercontent.com/RReverser/acorn-jsx/master/LICENSE"   } ] 
acorn-jsx@1.0.3: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://raw.githubusercontent.com/RReverser/acorn-jsx/master/LICENSE"   } ] 
acorn-jsx@2.0.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://raw.githubusercontent.com/RReverser/acorn-jsx/master/LICENSE"   } ] 
acorn-jsx@2.0.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://raw.githubusercontent.com/RReverser/acorn-jsx/master/LICENSE"   } ] 
acorn-jsx@3.0.0: expected: MIT, found: "MIT" null 
acorn-jsx@3.0.1: expected: MIT, found: "MIT" null 
async@0.1.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.10: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.11: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.12: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.13: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.14: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.15: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.16: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.17: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.18: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.19: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.20: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.21: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.22: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.3: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.4: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.5: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.6: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.7: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.8: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.1.9: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.10: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.3: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.4: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.5: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.6: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.7: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.8: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.2.9: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.3.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.4.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.4.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.5.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.6.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.6.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.6.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.7.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.8.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.9.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "https://github.com/caolan/async/raw/master/LICENSE"   } ] 
async@0.9.2: expected: MIT, found: "MIT" null 
colors@0.3.0: expected: MIT, found: null null 
colors@0.5.0: expected: MIT, found: null null 
colors@0.5.1: expected: MIT, found: null null 
colors@0.6.0: expected: MIT, found: null null 
colors@0.6.1: expected: MIT, found: null null 
colors@0.6.2: expected: MIT, found: null null 
enhanced-resolve@0.2.0: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.2.1: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.2.2: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.2.3: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.2.4: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.2.5: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.2.6: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.2.7: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.2.8: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.2.9: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.3.0: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.3.1: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.0: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.1: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.10: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.11: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.12: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.2: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.4: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.5: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.6: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.7: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.8: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.4.9: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.0: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.1: expected: MIT, found: "MIT" [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.10: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.11: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.12: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.13: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.14: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.15: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.3: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.4: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.5: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.6: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.7: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.8: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.5.9: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.6.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.6.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.7.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.7.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.7.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.7.3: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.7.4: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.7.5: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.7.6: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.8.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.8.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.8.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.8.3: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.8.4: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.8.5: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.8.6: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.9.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@0.9.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@2.0.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@2.0.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@2.0.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@2.1.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@2.2.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@2.2.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@2.2.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@2.3.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@3.0.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@3.0.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@3.0.2: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@3.0.3: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@3.1.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@3.2.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@3.3.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@3.4.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@3.4.1: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@4.0.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
enhanced-resolve@4.1.0: expected: MIT, found: null [   {     "type": "MIT",     "url": "http://www.opensource.org/licenses/mit-license.php"   } ] 
hat@0.0.0: expected: MIT, found: "MIT/X11" null 
hat@0.0.1: expected: MIT, found: "MIT/X11" null 
hat@0.0.2: expected: MIT, found: "MIT/X11" null 
```
</details>

Here's the command:
```sh
cat index.csv | awk '{split($0,a,","); print a[1] " "  a[2] " " a[3]}' | xargs -0 node tmp2 $@ | pbcopy
```

Here's the script:
```js
const { execSync } = require('child_process');

const [,,pkg,ver,lic] = process.argv;

if (pkg && !ver && !lic) {
	pkg.split('\n').forEach((line) => {
		const [pkg, ver, lic] = line.split(' ');
		work(pkg, ver, lic);
	});
} else {
	work(pkg, ver, lic);
}

function work(pkg, ver, lic) {
	const actual = String(execSync(`curl --silent -L https://unpkg.com/${pkg}@${ver}/package.json | jq '.license,.licenses'`));

	console.log(`${pkg}@${ver}: expected: ${lic}, found: ${actual.replace(/\n/g, ' ')}`);
}
```

kemitchell commented:
Peeled a couple packages off into #2 and #3. Let's get conf from the maintainers where we can.

ljharb commented:
Sure, but there's no need to confirm - the results are there ^

ljharb commented:
Do you want me to remove yargs and underscore from this PR, so it can be merged without waiting for #2 and #3?

ljharb commented:
rebased again.

ljharb commented:
I've rebased this again, and now that #2, #3, and #6 are all landed, this seems like it's OK to land? (after i remove the duplicates, which might end up emptying the content)

ljharb committed:
alpha-sort CSV

ljharb committed:
gitignore node_modules

ljharb commented:
The first commit alpha-sorts (and better semver-sorts most prereleases); the third is now adding packages and versions.

ljharb commented on index.csv:
```diff
@@ -194,7 +198,7 @@ slug,MIT,0.4.1,0.4.2,0.5.0,0.6.0,0.7.0,0.7.1,0.8.0,0.9.0,0.9.1,0.9.2,0.9.3
 socket.io,MIT,0.3.8,0.4.0,0.4.1,0.5.1,0.5.3,0.6.0,0.6.1,0.6.3,0.6.4,0.6.5,0.6.6,0.6.7,0.6.8,0.6.9,0.6.10,0.6.11,0.6.12,0.6.14,0.6.15,0.6.16,0.6.17,0.6.18
 source-map-loader,MIT,0.1.0,0.1.1,0.1.2,0.1.3,0.1.4,0.1.5,0.1.6,0.2.0,0.2.1,0.2.2,0.2.3
-source-map-support,MIT,0.2.9,0.2.10
+source-map-support,MIT,0.1.0,0.1.1,0.1.2,0.1.3,0.1.4,0.1.5,0.1.6,0.1.7,0.1.8,0.1.9,0.2.0,0.2.1,0.2.2,0.2.3,0.2.4,0.2.5,0.2.6,0.2.7,0.2.8,0.2.9,0.2.10,0.3.0,0.3.1,0.3.2
```
cc @evanw @julien-f @LinusU

LinusU replied:
These versions where before I joined the project so I'm afraid I can't help here...

ljharb commented on index.csv:
```diff
@@ -224,12 +228,12 @@ whatwg-fetch,MIT,0.5.0,0.6.0,0.6.1,0.7.0,0.8.0,0.8.1,0.8.2,0.9.0,0.10.0,0.10.1,0
 winston,MIT,0.7.3,0.8.0,0.8.1,0.8.2,0.8.3,0.9.0,1.0.0
 wordwrap,MIT,0.0.2
-wrench,MIT,0.1.0,1.0.0,1.1.0,1.2.0,1.3.0,1.3.1,1.3.2,1.3.3,1.3.4,1.3.5,1.3.6,1.3.7,1.3.8,1.3.9,1.4.0,1.4.1,1.4.2,1.4.3,1.4.4,1.5.0,1.5.1,1.5.3,1.5.4,1.5.5,1.5.6,1.5.7,1.5.8,1.5.9
+wrench,MIT,0.1.0,0.4.0,0.4.1,0.4.2,0.4.3,0.4.4,0.4.5,0.4.6,0.4.7,0.4.8,1.0.0,1.1.0,1.2.0,1.3.0,1.3.1,1.3.2,1.3.3,1.3.4,1.3.5,1.3.6,1.3.7,1.3.8,1.3.9,1.4.0,1.4.1,1.4.2,1.4.3,1.4.4,1.5.0,1.5.1,1.5.3,1.5.4,1.5.5,1.5.6,1.5.7,1.5.8,1.5.9
```
cc @ryanmcgrath

ryanmcgrath replied:
Wrench is deprecated and has been for years, and I'm not interested in updating it. What exactly is this for...?

ljharb replied:
It doesn't need any updates :-) all i'd need is your explicit confirmation that indeed, the MIT license should apply to all the listed versions of wrench.

ryanmcgrath replied:
If it's there and lasted the duration of the project, then yes, it applies. :)

ljharb commented on index.csv:
```diff
@@ -224,12 +228,12 @@ whatwg-fetch,MIT,0.5.0,0.6.0,0.6.1,0.7.0,0.8.0,0.8.1,0.8.2,0.9.0,0.10.0,0.10.1,0
 xmlbuilder,MIT,0.0.1,0.0.2,0.0.3,0.0.4,0.0.5,0.0.6,0.1.0,0.1.1,0.1.2,0.1.3,0.1.4,0.1.5,0.1.6,0.1.7,0.1.8,0.2.0,0.2.1,0.2.2,0.3.0,0.3.1,0.3.2,0.3.3,0.3.4,0.3.5,0.3.6,0.3.7,0.3.8,0.3.9,0.3.10,0.3.11,0.4.0,0.4.1,0.4.2,0.4.3,1.0.0,1.0.1,1.0.2,1.1.0,1.1.1,1.1.2,2.0.0,2.0.1,2.1.0,2.2.1,2.3.0,2.4.0,2.4.1,2.4.2
 xmlhttprequest,MIT,1.0.0,1.2.0,1.2.1,1.2.2,1.3.0,1.4.0,1.4.2,1.5.0,1.6.0,1.7.0
-xpath,MIT,0.0.1,0.0.2,0.0.4,0.0.5,0.0.6,0.0.7,0.0.8,0.0.9
+xpath,MIT,0.0.1,0.0.2,0.0.4,0.0.5,0.0.6,0.0.7,0.0.8,0.0.9,1.0.0,1.0.1,1.0.2,1.0.3,2.0.1,2.0.2,2.0.3,2.0.4,2.0.5,2.0.6,2.1.1,2.1.2,2.2.0,3.0.0,4.0.0
```
cc @JLRishe @goto100

ljharb commented on index.csv:
```diff
@@ -46,6 +48,7 @@ diff,BSD-3-Clause,1.4.0
 doctrine,Apache-2.0,1.5.0
 domutils,BSD-2-Clause,1.1.1,1.1.2,1.1.3,1.1.4,1.1.5,1.1.6,1.2.0,1.2.1,1.2.2,1.3.0,1.4.0,1.4.1,1.4.2,1.4.3,1.5.0,1.5.1
 elasticsearch,Apache-2.0,1.0.0,1.0.1,1.0.3,1.3.0,1.4.0,1.5.0,1.5.1,1.5.2,1.5.3,1.5.4,1.5.5,1.5.6,1.5.7,1.5.8,1.5.9,1.5.10,1.5.11,1.5.12,1.5.13,1.5.14,2.0.0,2.1.0,2.1.1,2.1.2,2.1.3,2.1.4,2.1.6,2.2.0,2.3.0,2.3.1,2.4.0,2.4.1,2.4.2,2.4.3,3.0.0,3.0.1,3.0.2,3.1.0,3.1.1,3.1.3,3.1.4,4.0.0,4.0.1,4.0.2,4.1.0,5.0.0
+enhanced-resolve,MIT,0.2.0,0.2.1,0.2.2,0.2.3,0.2.4,0.2.5,0.2.6,0.2.7,0.2.8,0.2.9,0.3.0,0.3.1,0.4.0,0.4.1,0.4.2,0.4.4,0.4.5,0.4.6,0.4.7,0.4.8,0.4.9,0.4.10,0.4.11,0.4.12,0.5.0,0.5.1,0.5.2,0.5.3,0.5.4,0.5.5,0.5.6,0.5.7,0.5.8,0.5.9,0.5.10,0.5.11,0.5.12,0.5.13,0.5.14,0.5.15,0.6.0,0.6.1,0.7.0,0.7.1,0.7.2,0.7.3,0.7.4,0.7.5,0.7.6,0.8.0,0.8.1,0.8.2,0.8.3,0.8.4,0.8.5,0.8.6,0.9.0,0.9.1,2.0.0,2.0.1,2.0.2,2.1 ...
```
cc @sokra

ljharb commented on index.csv:
```diff
@@ -27,6 +28,7 @@ cli,MIT,0.1.0,0.1.1,0.1.3,0.1.4,0.1.5,0.1.6,0.1.7,0.1.8,0.1.9,0.2.0,0.2.1-1,0.2.
 clui,MIT,0.1.2,0.1.3,0.2.0,0.2.1,0.3.1,0.3.3,0.3.4,0.3.5,0.3.6
 codemirror,MIT,2.33.0,3.11.1,3.12.0,3.13.0,3.14.1,3.15.0,3.16.0,3.17.0,3.18.0,3.19.0,3.20.0,3.21.0,3.22.0,3.23.0,3.24.0,4.0.3,4.1.0,4.2.0,4.3.0,4.4.0,4.5.0,4.6.0,4.7.0,4.8.0,4.9.0,4.10.0,4.11.0,4.12.0,4.13.0,5.0.0,5.1.0,5.2.0
 coffee-script,MIT,0.7.0,0.7.1,0.7.2,0.9.0,0.9.1,0.9.2,0.9.3,0.9.4,0.9.5,0.9.6,1.0.0,1.0.1,1.1.0,1.1.1,1.1.2,1.1.3,1.2.0,1.3.0,1.3.1,1.3.2,1.3.3,1.4.0,1.5.0,1.6.0,1.6.1,1.6.2,1.6.3
+colors,MIT,0.3.0,0.5.0,0.5.1,0.6.0,0.6.1,0.6.2
```
cc @marak

ljharb commented on index.csv:
```diff
@@ -1,11 +1,12 @@
 archiver,MIT,0.1.0,0.1.1,0.2.0,0.2.1,0.2.2,0.3.0,0.4.0,0.4.1,0.4.2,0.4.3,0.4.4,0.4.5,0.4.6,0.4.7,0.4.8,0.4.9,0.4.10,0.5.0-alpha,0.5.0,0.5.1,0.5.2,0.6.0,0.6.1,0.7.0,0.7.1,0.8.0,0.8.1,0.9.0,0.9.1,0.10.0-alpha,0.10.0,0.10.1,0.11.0-alpha,0.11.0,0.12.0,0.13.0,0.13.1,0.14.0,0.14.1,0.14.2,0.14.3
 argparse,MIT,0.1.0,0.1.1,0.1.2,0.1.3,0.1.4,0.1.5,0.1.6,0.1.7,0.1.8,0.1.9,0.1.10,0.1.11,0.1.12,0.1.13,0.1.14,0.1.15
-async,MIT,0.1.0,0.1.1,0.1.2,0.1.3,0.1.4,0.1.5,0.1.6,0.1.7,0.1.8,0.1.9,0.1.10,0.1.11,0.1.12,0.1.13,0.1.14,0.1.15,0.1.16,0.1.17,0.1.18,0.1.19,0.1.20,0.1.21,0.1.22,0.2.0,0.2.1,0.2.2,0.2.3,0.2.4,0.2.5,0.2.6,0.2.7,0.2.8,0.2.9,0.2.10,0.3.0,0.4.0,0.4.1,0.5.0,0.6.0,0.6.1,0.6.2,0.7.0,0.8.0,0.9.0
+async,MIT,0.1.0,0.1.1,0.1.2,0.1.3,0.1.4,0.1.5,0.1.6,0.1.7,0.1.8,0.1.9,0.1.10,0.1.11,0.1.12,0.1.13,0.1.14,0.1.15,0.1.16,0.1.17,0.1.18,0.1.19,0.1.20,0.1.21,0.1.22,0.2.0,0.2.1,0.2.2,0.2.3,0.2.4,0.2.5,0.2.6,0.2.7,0.2.8,0.2.9,0.2.10,0.3.0,0.4.0,0.4.1,0.5.0,0.6.0,0.6.1,0.6.2,0.7.0,0.8.0,0.9.0,0.9.2
```
cc @caolan

ljharb commented on index.csv:
```diff
@@ -1,11 +1,12 @@
+acorn-jsx,MIT,0.6.1,1.0.0,1.0.1,1.0.2,1.0.3,2.0.0,2.0.1,3.0.0,3.0.1
```
cc @rreverser @adrianheine @marijn

RReverser replied:
Previous versions used `licenses` declaration, which is now deprecated, but was still correct SPDX declaration at the time. https://unpkg.com/browse/acorn-jsx@1.0.0/package.json

Might be worth updating the script you used for collecting the data to count this old style as valid too.

ljharb reviewed (commented):
I've cc'd the maintainers I could find, hopefully each of yall can confirm that these license corrections are valid.

kemitchell commented:
@ljharb appreciate energy on this. With the sort, this is going to become fairly well unreviewable on my end.

When you've done everything you practically can to make sure the patch contains _the truth, the whole truth, and nothing but the truth_, land and publish. You already have registry perms.

ljharb commented:
@kemitchell yeah it’s best to review by commit :-) i can pull the sort in directly or via another PR, tho.

kemitchell commented:
@ljharb don't bother. I'm not going to catch up with you. Just get it to where you're confident, merge, and publish.
